crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -1537,7 +1550,9 @@ struct ProcessConfigsResult {
 /// Trait used by [`ScheduleGraph::process_configs`] to process a single [`NodeConfig`].
 trait ProcessNodeConfig: Sized {
     /// Process a single [`NodeConfig`].
-    fn process_config(schedule_graph: &mut ScheduleGraph, config: NodeConfig<Self>) -> NodeId;
+    fn process_config(schedule_graph: &mut ScheduleGraph, config: NodeConfig<Self>) -> NodeId
+    where
+        Self: NodeType;
```

Comment:
Since `ProcessNodeConfig` only has this single function, the `NodeType` bound should be instead moved to the trait definition like so:

```rust
trait ProcessNodeConfig: NodeType + Sized {
```

I can't directly suggest it unfortunately because GitHub won't let me.